src/app/nav/header/header.component.ts

```diff
@@ -19,9 +19,6 @@ export class HeaderComponent implements OnInit {
 
   ngOnInit() {
     this.isLoggedIn = this.authService.isLoggedIn();
-    if (!this.isLoggedIn) {
-      this.router.navigate(['/login']);
```

Comment:
this implementation needs to be moved else where. Since we still need to redirect users to the login pages if they try to access protected pages without logging in

Reply:
Actually, the `isLoggedIn()` function is the one updating the isLoggedIn() variable  which the logout button depends on to be displayed so all routes are protected in sidemenu.component.ts where we have 
```
    this.loginuser = JSON.parse(sessionStorage.getItem('loginuser')!);
    if (!this.loginuser) {
      return this.logout();
    }
```
 and the logout function redirects